rest/src/main/kotlin/service/ChannelService.kt

```diff
@@ -233,6 +241,100 @@ class ChannelService(requestHandler: RequestHandler) : RestService(requestHandle
             keys[Route.ChannelId] = channelId
             body(ChannelFollowRequest.serializer(), request)
         }
+    suspend fun startPublicThread(
+        channelId: Snowflake,
+        messageId: Snowflake,
+        request: StartThreadRequest
+    ): DiscordChannel {
+        return call(Route.StartPublicThreadPost) {
+            keys[Route.ChannelId] = channelId
+            keys[Route.MessageId] = messageId
+            body(StartThreadRequest.serializer(), request)
+        }
+    }
+
+
+    suspend fun startPrivateThread(
+        channelId: Snowflake,
+        request: StartThreadRequest
+    ): DiscordChannel {
+        return call(Route.StartPrivateThreadPost) {
+            keys[Route.ChannelId] = channelId
+            body(StartThreadRequest.serializer(), request)
+        }
+    }
```

Comment:
These functions are incorrectly named, they don't create public and private threads respectively.

Reply:
What do you mean?